app/lib/contextual-links.coffee

```diff
@@ -0,0 +1,435 @@
+#   /projects/project_owner/project_name/talk/recents/user_name           -> All recent comments for user_name made to the Talk of project_name
+#   /projects/project_owner/project_name/talk/recents                     -> All recent comments for all users and made to the Talk of project_name
+#   /talk/recents/user_name                                               -> All recent comments made by user_name to all projects
+#   /talk/recents/board_id                                                -> All recent comments on board_id made by user_name
+#   /talk/recents/                                                        -> All recent comments for all users and all projects
+#
+
+
+module.exports =
+
+  
+  #  In general, when we are in a user context, that could be the user who we're logged in as or the user whose content 
+  # we're viewing. Therefore we can deduce this from props 
+  # This method will return the login name of the user whose context we are in
+  # This comes from the collection, if present, otherwise from the logged in user if there is one.
+
+  getContextUserLogin: (props) ->
+    if props.favorite and props.params?.favorites_owner?
+      return props.params.favorites_owner
+    else if !props.favorite and props.params?.collection_owner?
+      return props.params.collection_owner
+    else if !props.project? and props.params?.owner?
+      return props.params.owner
+    else if props.user?
+      return props.user.login
+    else
+      return props.params.owner # which may be null
+  
+  # to discourage navigation links from ending up with a line break on small screens, we can swap all spaces for &nbsp;
+  makeTextUnbreakable: (text) ->
+    text.replace /\ /g, "\u00a0"
+    
+  # ensures that a link will stay in the project context if we are in the context of a project
+  prefixLinkIfNeeded: (props,url) ->
+    if props.project?
+      # keeps project in context
+      url = "/projects/#{props.project.slug}" + url
+    return url
+
+  # determines current user/project/user+project filters in effect from the URL.
+  # returns the correct format for directly attaching to the query.
+  getFiltersFromPath: (props) ->
+    filters = {}
+    pathParts = @safelyGetPath(props)
+    [firstPart, ..., lastPart] = pathParts
+
+    if firstPart == "projects" and pathParts.length < 6 and lastPart != "all"
+      filters["project_ids"] = props.project.id
+    if firstPart == "collections" and pathParts.length == 2 and pathParts[1] != "" and pathParts[1] != "all"
+      filters["owner"] = pathParts[1]
+    if firstPart == "favorites" and pathParts.length == 2 and pathParts[1] != "" and pathParts[1] != ""
+      filters["owner"] = pathParts[1]
+    if pathParts.length>4 and pathParts[3] == "collections" and pathParts[4] != "" and pathParts[4] != "all"
+      filters["owner"] = pathParts[4]
+    if pathParts.length>4 and pathParts[3] == "favorites" and pathParts[4] != "" and pathParts[4] != "all"
+      filters["owner"] = pathParts[4]
```

Comment:
If this is going to be in coffeescript, could you be consistent with using its syntax, such as using `is` instead of `==` and `isnt` instead of `!=`. In javascript, as an aside, you should use `===` or `!==`.